optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/rostergrid/view/SubLaneView.java

```diff
@@ -75,31 +77,36 @@ public void destroy() {
     }
 
     @EventHandler("sub-lane")
-    public void onClick(final @ForEvent("click") MouseEvent e) {
+    public void onMouseDown(final @ForEvent("mousedown") MouseEvent e) {
+        hasMouseMoved = false;
+    }
 
-        if (!e.target.equals(e.currentTarget)) {
-            return;
+    @EventHandler("sub-lane")
+    public void onMouseMove(final @ForEvent("mousemove") MouseEvent e) {
+        hasMouseMoved = true;
+    }
+
+    @EventHandler("sub-lane")
+    public void onMouseUp(final @ForEvent("mouseup") MouseEvent e) {
+        if (!hasMouseMoved) {
+            onClick(e);
```

Comment:
If a "click" fires after a "mouseup", then are you calling onClick twice per click? Also why don't we want to call onClick if the mouse moves during the click?

Reply:
OnClick once - when the user releases the click. The issue lies when dragging/resizing shifts (only available in rotation currently) - often after dragging/resizing the shift, either 1) A new shift is added or 2) The shift edit dialog pop ups. Niether is desired; by ensuring the mouse didn't move during the click, we can ensure that is what the user desired to do. Dragging/Resizing will eventually be added to the Roster Screens.

Reply:
Right, I was getting confused by the `@ForEvent` annotation on your onClick parameter. Could you remove that?

Reply:
Thanks for finding that! Fixed